mlflow/store/sqlalchemy_store.py

```diff
@@ -342,37 +342,9 @@ def log_metric(self, run_uuid, metric):
         with self.ManagedSessionMaker() as session:
             run = self._get_run(run_uuid=run_uuid, session=session)
             self._check_run_is_active(run)
-
-            try:
-                self._get_or_create(model=SqlMetric, run_uuid=run_uuid, key=metric.key,
-                                    value=metric.value, timestamp=metric.timestamp, session=session)
-                # Explicitly commit the session in order to catch potential integrity errors
-                # while maintaining the current managed session scope ("commit" checks that
-                # a transaction satisfies uniqueness constraints and throws integrity errors
-                # when they are violated; "get_or_create()" does not perform these checks). It is
-                # important that we maintain the same session scope because, in the case of
-                # an integrity error, we want to examine the uniqueness of metric (timestamp, value)
-                # tuples using the same database state that the session uses during "commit".
-                # Creating a new session synchronizes the state with the database. As a result, if
-                # the conflicting (timestamp, value) tuple were to be removed prior to the creation
-                # of a new session, we would be unable to determine the cause of failure for the
-                # first session's "commit" operation.
-                session.commit()
-            except sqlalchemy.exc.IntegrityError:
-                # Roll back the current session to make it usable for further transactions. In the
-                # event of an error during "commit", a rollback is required in order to continue
-                # using the session. In this case, we re-use the session because the SqlRun, `run`,
-                # is lazily evaluated during the invocation of `run.metrics`.
-                session.rollback()
-                existing_metric = [m for m in run.metrics
-                                   if m.key == metric.key and m.timestamp == metric.timestamp]
-                if len(existing_metric) > 0:
-                    m = existing_metric[0]
-                    raise MlflowException(
-                        "Metric={} must be unique. Metric already logged value {}"
-                        " at {}".format(metric, m.value, m.timestamp), INVALID_PARAMETER_VALUE)
-                else:
-                    raise
+            # ToDo: Consider prior checks for null, type, metric name validations, ... etc.
+            self._get_or_create(model=SqlMetric, run_uuid=run_uuid, key=metric.key,
```

Comment:
Adding `value` to PK constraint is a good idea.

Like the cleanup here. Neat! Is there any other possibility of an `IntegrityError` which would get exposed to client?

Reply:
The `IntegrityError` will be caught and reraised as an `MlflowException` (see https://github.com/mlflow/mlflow/blob/a7272b7309b6d53e70afe7c19dc3ca383962261a/mlflow/store/sqlalchemy_store.py#L89)

Reply:
Gotchya! Thanks...